executor/executor_simple.go

```diff
@@ -371,7 +372,7 @@ func (e *SimpleExec) executeDropUser(s *ast.DropUserStmt) error {
 	}
 	if len(failedUsers) > 0 {
 		errMsg := "Operation DROP USER failed for " + strings.Join(failedUsers, ",")
-		return errors.New(errMsg)
+		return terror.ClassExecutor.New(mysql.ErrCannotUser, errMsg)
```

Comment:
You need add this error code to the map of ErrClassToMySQLCodes, you can refer to the function of init in table.go